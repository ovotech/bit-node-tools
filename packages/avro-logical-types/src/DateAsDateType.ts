import { Type, types } from 'avsc';
import { toDate, fromDate } from './helpers/epoch-days';

/**
 * Custom logical type used to encode native Date objects as int.
 */
export class DateAsDateType extends types.LogicalType {
  _fromValue(val: number) {
    return toDate(val);
  }
  _toValue(date: any) {
    return fromDate(date);
  }
  _resolve(type: any) {
    if (Type.isType(type, 'int', 'logical:date')) {
      return this._fromValue;
    }
    return undefined;
  }
}
