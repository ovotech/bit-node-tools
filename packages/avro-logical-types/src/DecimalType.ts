import { Type, types } from 'avsc';
import { Decimal } from 'decimal.js';
import { Int64BE } from 'int64-buffer';

export class DecimalType extends types.LogicalType {
  public precision: number;
  public scale: number;

  constructor(schema: any, opts?: any) {
    super(schema, opts);
    this.precision = schema.precision;
    this.scale = schema.scale;
  }

  public _export(attrs: any) {
    attrs.precision = this.precision;
    attrs.scale = this.scale;
  }

  public _resolve(type: Type) {
    if (
      type instanceof DecimalType &&
      Type.isType(type, 'logical:decimal') &&
      type.precision === this.precision &&
      type.scale === this.scale
    ) {
      return <T>(x: T): T => x;
    }
  }

  public _fromValue(buf: any) {
    if (!(buf instanceof Buffer)) {
      throw new Error('expecting underlying Buffer type');
    }

    if (buf.length > 8) {
      throw new Error('buffers with more than 64bits are not supported');
    }

    return bufferToDecimal(buf, this.scale);
  }

  public _toValue(val: any) {
    if (!(val instanceof Decimal)) {
      throw new Error('expecting Decimal type');
    }

    return decimalToBuffer(val, this.scale);
  }
}

export function decimalToBuffer(val: Decimal, scale: number): Buffer {
  const scaled = val.mul(10 ** scale);
  const value = new Int64BE(scaled.trunc().toString());
  return value.toBuffer();
}

export function bufferToDecimal(buf: Buffer, scale: number): Decimal {
  const unscaled = new Int64BE(to64Bit(buf));
  return new Decimal(unscaled.toString()).div(10 ** scale);
}

export function to64Bit(buf: Buffer) {
  // tslint:disable-next-line:no-bitwise
  const isNegatve = buf[0] & 0x80;
  const toFill = 8 - buf.length;

  const bufferElements = Array.prototype.slice.call(buf, 0);

  return Buffer.from([...[...new Array(toFill)].map(_ => (isNegatve ? 0xff : 0x00)), ...bufferElements]);
}
