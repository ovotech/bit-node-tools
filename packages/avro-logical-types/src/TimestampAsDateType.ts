import { Type, types } from 'avsc';
/**
 * Custom logical type used to encode native Date objects as long.
 */
export class TimestampAsDateType extends types.LogicalType {
  _fromValue(val: number) {
    return new Date(val);
  }
  _toValue(date: any) {
    return date instanceof Date ? date.getTime() : date;
  }
  _resolve(type: any) {
    if (Type.isType(type, 'long', 'logical:timestamp-millis')) {
      return this._fromValue;
    }
    return undefined;
  }
}
