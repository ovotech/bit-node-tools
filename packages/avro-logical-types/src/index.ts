export { DateType } from './DateType';
export { DateAsDateType } from './DateAsDateType';
export { DecimalType } from './DecimalType';
export { TimestampType } from './TimestampType';
export { TimestampAsDateType } from './TimestampAsDateType';
