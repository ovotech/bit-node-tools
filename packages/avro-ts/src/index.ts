import { Schema, schema } from 'avsc';
import * as ts from 'typescript';

export interface Registry {
  [key: string]: ts.InterfaceDeclaration;
}

type UnionRegistry = {
  [key: string]: string[];
};

export interface Context {
  recordAlias: string;
  namesAlias: string;
  namespacedPrefix: string;
  registry: Registry;
  unionRegistry: UnionRegistry;
  unionMember: boolean;
  namespace?: string;
  namespaces: { [key: string]: ts.TypeReferenceNode };
  logicalTypes: { [key: string]: ts.TypeReferenceNode };
  visitedLogicalTypes: Array<string>;
}

export interface Result<TsType = ts.TypeNode> {
  type: TsType;
  context: Context;
}

type TypeRecordType = { type: schema.RecordType };

export type Convert<TType = Schema> = (context: Context, type: TType) => Result<any>;

export const result = <TsType = ts.TypeNode>(context: Context, type: TsType): Result<TsType> => ({
  context,
  type,
});

export const mapContext = <T = any, TsType = ts.TypeNode>(
  context: Context,
  items: T[],
  callbackfn: (context: Context, item: T) => Result<TsType>,
) =>
  items.reduce<{ items: TsType[]; context: Context }>(
    (all, item) => {
      const current = callbackfn(all.context, item);
      return {
        items: [...all.items, current.type],
        context: current.context,
      };
    },
    { items: [], context },
  );

export const withContexts = (context: Context, items: Context[]) =>
  items.reduce(
    (all, itemContext) => ({
      ...all,
      registry: { ...all.registry, ...itemContext.registry },
      namespaces: { ...all.namespaces, ...itemContext.namespaces },
    }),
    context,
  );

export const withEntry = (context: Context, entry: ts.InterfaceDeclaration): Context => ({
  ...context,
  registry: { ...context.registry, [entry.name.text]: entry },
});

export const withNamespace = (context: Context, record: schema.RecordType): Context => ({
  ...context,
  namespace: record.namespace,
  namespaces: {
    ...context.namespaces,
    [fullyQualifiedName(context, record)]: ts.createTypeReferenceNode(record.name, undefined),
  },
});

export interface State {
  output: string[];
  repository: { [key: string]: string };
  logicalTypes: { [key: string]: string };
}

const docToJSDoc = (doc: string) =>
  `*\n${doc
    .split('\n')
    .map(line => ` * ${line}`)
    .join('\n')}\n `;

const convertRecord: Convert<schema.RecordType> = (context, type) => {
  const namespaceContext = type.namespace ? withNamespace(context, type) : context;
  const fieldContext = { ...namespaceContext, unionMember: false };

  const fields = type.fields.map(fieldType => {
    const field = convertType(fieldContext, fieldType.type);
    const prop = ts.createPropertySignature(
      undefined,
      fieldType.name,
      isOptional(fieldType.type) ? ts.createToken(ts.SyntaxKind.QuestionToken) : undefined,
      field.type,
      undefined,
    );

    const propWithDoc = fieldType.doc
      ? ts.addSyntheticLeadingComment(prop, ts.SyntaxKind.MultiLineCommentTrivia, docToJSDoc(fieldType.doc), true)
      : prop;

    return result(field.context, propWithDoc);
  });

  const interfaceType = ts.createInterfaceDeclaration(
    undefined,
    [ts.createToken(ts.SyntaxKind.ExportKeyword)],
    type.name,
    undefined,
    undefined,
    fields.map(field => field.type),
  );
  const recordContext = withContexts(withEntry(fieldContext, interfaceType), fields.map(item => item.context));

  if (context.unionMember) {
    const namespaced = fullyQualifiedName(context, type);
    const currentNamespace = type.namespace || context.namespace;
    const props = [
      ts.createPropertySignature(
        undefined,
        ts.createStringLiteral(namespaced),
        undefined,
        ts.createTypeReferenceNode(type.name, undefined),
        undefined,
      ),
    ];

    if (currentNamespace) {
      props.push(
        ...(context.unionRegistry[currentNamespace] || [])
          .filter((name: string) => name !== type.name)
          .map(name =>
            ts.createPropertySignature(
              undefined,
              ts.createStringLiteral(`${currentNamespace}.${name}`),
              ts.createToken(ts.SyntaxKind.QuestionToken),
              ts.createKeywordTypeNode(ts.SyntaxKind.NeverKeyword),
              undefined,
            ),
          ),
      );
    }

    const namespacedInterfaceType = ts.createInterfaceDeclaration(
      undefined,
      [ts.createToken(ts.SyntaxKind.ExportKeyword)],
      `${context.namespacedPrefix}${type.name}`,
      undefined,
      undefined,
      props,
    );

    return result(
      withEntry(recordContext, namespacedInterfaceType),
      ts.createTypeReferenceNode(namespacedInterfaceType.name.text, undefined),
    );
  }
  return result(recordContext, ts.createTypeReferenceNode(type.name, undefined));
};

const convertType: Convert = (context, type) => {
  if (typeof type === 'string') {
    return convertPredefinedType(context, type);
  } else if (Array.isArray(type)) {
    return convertArrayType(context, type);
  } else if (isLogicalType(type)) {
    return convertLogicalType(context, type);
  } else if (isRecordType(type)) {
    return convertRecord(context, type);
  } else if (isArrayType(type)) {
    const itemType = convertType(context, type.items);
    return result(itemType.context, ts.createArrayTypeNode(itemType.type));
  } else if (isMapType(type)) {
    return convertMapType(context, type);
  } else if (isEnumType(type)) {
    return convertEnum(context, type);
  } else {
    throw new Error(`Cannot work out type ${type}`);
  }
};

const convertPrimitive: Convert = (context, avroType) => {
  switch (avroType) {
    case 'long':
    case 'int':
    case 'double':
    case 'float':
      return result(context, ts.createKeywordTypeNode(ts.SyntaxKind.NumberKeyword));
    case 'bytes':
      return result(context, ts.createTypeReferenceNode('Buffer', undefined));
    case 'null':
      return result(context, ts.createNull());
    case 'boolean':
      return result(context, ts.createKeywordTypeNode(ts.SyntaxKind.BooleanKeyword));
    case 'string':
      return result(context, ts.createKeywordTypeNode(ts.SyntaxKind.StringKeyword));
    default:
      return result(context, ts.createKeywordTypeNode(ts.SyntaxKind.AnyKeyword));
  }
};

const convertEnum: Convert<schema.EnumType> = (context, enumType) =>
  result(
    context,
    ts.createUnionTypeNode(enumType.symbols.map(symbol => ts.createLiteralTypeNode(ts.createLiteral(symbol)))),
  );

const convertLogicalType: Convert<schema.LogicalType> = (context, type) => {
  if (context.logicalTypes[type.logicalType]) {
    if (!context.visitedLogicalTypes.includes(type.logicalType)) context.visitedLogicalTypes.push(type.logicalType);
    return result(context, context.logicalTypes[type.logicalType]);
  }
  return convertPrimitive(context, type.type);
};

const convertPredefinedType: Convert<string> = (context, type) =>
  context.namespaces[type] ? result(context, context.namespaces[type]) : convertPrimitive(context, type);

const convertArrayType: Convert<any[]> = (context, type) => {
  const map = mapContext(context, type, (itemContext, item) => {
    if (typeof item === 'object' && !Array.isArray(item) && isRecordType(item)) {
      return convertType({ ...itemContext, unionMember: true }, item);
    } else {
      return convertType(itemContext, item);
    }
  });

  return result(map.context, ts.createUnionTypeNode(map.items));
};

const convertMapType: Convert<schema.MapType> = (context, type) => {
  const map = convertType(context, type.values);
  return result(
    map.context,
    ts.createTypeLiteralNode([
      ts.createIndexSignature(
        undefined,
        undefined,
        [
          ts.createParameter(
            undefined,
            undefined,
            undefined,
            'index',
            undefined,
            ts.createKeywordTypeNode(ts.SyntaxKind.StringKeyword),
            undefined,
          ),
        ],
        map.type,
      ),
    ]),
  );
};

const isRecordType = (type: Schema): type is schema.RecordType =>
  typeof type === 'object' && 'type' in type && type.type === 'record';

const isArrayType = (type: Schema): type is schema.ArrayType =>
  typeof type === 'object' && 'type' in type && type.type === 'array';

const isMapType = (type: Schema): type is schema.MapType =>
  typeof type === 'object' && 'type' in type && type.type === 'map';

const isEnumType = (type: Schema): type is schema.EnumType =>
  typeof type === 'object' && 'type' in type && type.type === 'enum';

const isLogicalType = (type: Schema): type is schema.LogicalType => typeof type === 'object' && 'logicalType' in type;

const isUnion = (type: Schema): type is schema.NamedType[] => typeof type === 'object' && Array.isArray(type);

const isRecordParent = (type: any): type is TypeRecordType => typeof type === 'object' && typeof type.type === 'object';

const isUnionParent = (type: any): type is { type: Array<schema.AvroSchema> } => Array.isArray(type.type);

const isOptional = (type: Schema): boolean => {
  if (isUnion(type)) {
    const t1 = type[0];
    if (typeof t1 === 'string') {
      return t1 === 'null';
    }
  }
  return false;
};

const fullyQualifiedName = (context: Context, type: schema.RecordType) => {
  const currentNamespace = type.namespace || context.namespace;
  return currentNamespace ? `${currentNamespace}.${type.name}` : type.name;
};

const printAstNode = (nodes: Array<ts.Node>, { importLines }: { importLines: Array<string> }): string => {
  const resultFile = ts.createSourceFile('someFileName.ts', '', ts.ScriptTarget.Latest);
  const printer = ts.createPrinter({ newLine: ts.NewLineKind.LineFeed });

  return importLines.concat(nodes.map(n => printer.printNode(ts.EmitHint.Unspecified, n, resultFile))).join('\n\n');
};

type LogicalTypeWithImport = { import: string; type: string };
type LogicalTypeDefinition = string | LogicalTypeWithImport;

type AvroTsOptions = {
  logicalTypes?: { [key: string]: LogicalTypeDefinition };
  recordAlias?: string;
  namespacedPrefix?: string;
  namesAlias?: string;
};
const defaultOptions = {
  recordAlias: 'Record',
  namespacedPrefix: 'Namespaced',
  namesAlias: 'Names',
};

export function avroTs(recordType: schema.RecordType, options: AvroTsOptions = {}): string {
  const logicalTypes = options.logicalTypes || {};
  const isRootUnion = Array.isArray(recordType);

  const context: Context = {
    ...options,
    recordAlias: options.recordAlias || defaultOptions.recordAlias,
    namesAlias: options.namesAlias || defaultOptions.namesAlias,
    namespacedPrefix: options.namespacedPrefix || defaultOptions.namespacedPrefix,
    unionMember: isRootUnion,
    registry: {},
    unionRegistry: buildUnionRegistry({}, recordType, { namespace: recordType.namespace, unionMember: isRootUnion }),
    namespaces: {},
    visitedLogicalTypes: [],
    logicalTypes: Object.entries(logicalTypes).reduce((all, [name, type]) => {
      const typeStr = (type as LogicalTypeWithImport).type ? (type as LogicalTypeWithImport).type : (type as string);
      return {
        ...all,
        [name]: ts.createTypeReferenceNode(typeStr, undefined),
      };
    }, {}),
  };

  const mainNode = convertType(context, recordType);

  const importLines = context.visitedLogicalTypes
    .map(visitedType => (logicalTypes[visitedType] as LogicalTypeWithImport).import)
    .filter(Boolean);

  const nodes: Array<ts.Node> = [
    ts.createTypeAliasDeclaration(
      undefined,
      [ts.createToken(ts.SyntaxKind.ExportKeyword)],
      context.recordAlias,
      undefined,
      mainNode.type,
    ) as ts.Node,
  ].concat(Object.values(mainNode.context.registry));

  const namesNamespace = unionRegisryToNamespace(context.unionRegistry, context.namesAlias);

  if (namesNamespace) {
    nodes.unshift(namesNamespace);
  }

  return printAstNode(nodes, { importLines });
}

function unionRegisryToNamespace(registry: UnionRegistry, namespaceName: string): ts.Node | undefined {
  const names = Object.keys(registry).reduce<Array<ts.Statement>>(
    (nodes, namespace) =>
      nodes.concat(
        registry[namespace].map(name =>
          ts.createVariableStatement(
            [ts.createToken(ts.SyntaxKind.ExportKeyword)],
            ts.createVariableDeclarationList(
              [ts.createVariableDeclaration(name, undefined, ts.createLiteral(`${namespace}.${name}`))],
              ts.NodeFlags.Const,
            ),
          ),
        ),
      ),
    [],
  );

  if (!names.length) {
    return;
  }

  const nsNode = ts.createModuleDeclaration(
    [],
    [ts.createToken(ts.SyntaxKind.ExportKeyword)],
    ts.createIdentifier(namespaceName),
    ts.createModuleBlock(names),
    ts.NodeFlags.Namespace,
  );
  return nsNode;
}

function buildUnionRegistry(
  registry: UnionRegistry,
  schema: schema.AvroSchema,
  context: { namespace?: string; unionMember: boolean },
): UnionRegistry {
  if (Array.isArray(schema)) {
    return schema.reduce((acc, schema) => buildUnionRegistry(acc, schema, context), registry);
  }

  if (isRecordParent(schema)) {
    return buildUnionRegistry(registry, schema.type, { ...context, unionMember: isUnionParent(schema) });
  }

  if (isRecordType(schema)) {
    const { name, fields } = schema;
    const currentNamespace = schema.namespace || context.namespace;
    if (currentNamespace && context.unionMember) {
      (registry[currentNamespace] = registry[currentNamespace] || []).push(name);
    }

    fields.reduce(
      (acc, field) =>
        // @ts-ignore This works, but a bit too dynamic for TS
        buildUnionRegistry(acc, field, {
          ...context,
          unionMember: false,
          namespace: schema.namespace || context.namespace,
        }),
      registry,
    );
  }

  return registry;
}
