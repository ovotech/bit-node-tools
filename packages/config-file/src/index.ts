import { existsSync, readFileSync } from 'fs';

export const resolveValues = <T>(config: Partial<T>, env: NodeJS.ProcessEnv) => {
  const keys = Object.keys(config) as Array<keyof T>;

  return keys.reduce(
    (acc, key) => {
      const value = config[key];
      const resolvedValue =
        typeof value === 'string'
          ? value.replace(/\$\{([A-Za-z][A-Za-z0-9\_]*)\}/g, (match, name, offset, str) => String(env[name]))
          : value;

      return { ...acc, [key]: resolvedValue };
    },
    {} as T,
  );
};

export interface LoadConfigFileOptions<T> {
  env: NodeJS.ProcessEnv;
  file: string;
  defaults?: Partial<T>;
  required?: Array<keyof T>;
}

export const loadConfigFile = <T>({ env, file, defaults = {}, required = [] }: LoadConfigFileOptions<T>): T => {
  if (!existsSync(file)) {
    throw new Error(`Configuration file ${file} not found.`);
  }

  const userConfig: Partial<T> = JSON.parse(String(readFileSync(file)));

  const missingKeys = required.filter(key => !Object.keys(userConfig).includes(String(key)));
  if (missingKeys.length) {
    throw new Error(`Configuration should include ${missingKeys.map(key => `"${String(key)}"`).join(', ')} in ${file}`);
  }
  const config = resolveValues(userConfig, env);

  return { ...defaults, ...config };
};
