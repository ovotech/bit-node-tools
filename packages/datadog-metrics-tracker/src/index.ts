var StatsD = require("hot-shots");

export { MetricsTracker } from './base';
export { ExternalRequestMetricsTracker } from './external-request';
export { KafkaMetricsTracker, ProcessingState } from './kafka';
export { ResponseMetricsTracker } from './response';

export interface DataDogConfig extends NodeJS.ProcessEnv {
  DATADOG_HOST?: string;
  DATADOG_PORT?: string;
}

export const createDataDogConnection = ({DATADOG_HOST = '', DATADOG_PORT = '15661'}: DataDogConfig) => {
  var client  = new StatsD({
    port: DATADOG_PORT,
    host: DATADOG_HOST,
    errorHandler: (error:Error) => {
      console.error(error);
    }
  });

  return client ;
};
