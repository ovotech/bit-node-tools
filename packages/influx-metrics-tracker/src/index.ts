import { InfluxDB, ISingleHostConfig } from 'influx';

export { MetricsTracker } from './base';
export { ExternalRequestMetricsTracker } from './external-request';
export { KafkaMetricsTracker } from './kafka';
export { ResponseMetricsTracker } from './response';

export interface InfluxConfig extends NodeJS.ProcessEnv {
  INFLUXDB_HOST?: string;
  INFLUXDB_DATABASE?: string;
  INFLUXDB_PORT?: string;
  INFLUXDB_USER?: string;
  INFLUXDB_PASSWORD?: string;
}

export const createInfluxConnection = ({
  INFLUXDB_HOST = '',
  INFLUXDB_DATABASE = 'default-db',
  INFLUXDB_PORT = '15661',
  INFLUXDB_USER = '',
  INFLUXDB_PASSWORD = '',
}: InfluxConfig): InfluxDB => {
  const config: ISingleHostConfig = {
    host: INFLUXDB_HOST,
    database: INFLUXDB_DATABASE,
    port: parseInt(INFLUXDB_PORT, 10),
    username: INFLUXDB_USER,
    password: INFLUXDB_PASSWORD,
    protocol: 'https',
  };
  return new InfluxDB(config);
};
