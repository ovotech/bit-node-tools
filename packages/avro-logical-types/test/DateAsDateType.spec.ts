import { Schema, Type } from 'avsc';
import { DateAsDateType } from '../src';

describe('Unit test', () => {
  it('DateAsDateType', async () => {
    const type = Type.forSchema(
      {
        type: 'record',
        fields: [{ name: 'startedAt', type: { type: 'int', logicalType: 'date' } }],
      } as Schema,
      { logicalTypes: { date: DateAsDateType } },
    );

    const values = [{ startedAt: new Date('2018-06-05') }, { startedAt: 17687 }];
    const expected = [
      { startedAt: new Date('2018-06-05T00:00:00.000Z') },
      { startedAt: new Date('2018-06-05T00:00:00.000Z') },
    ];

    const buffers = values.map(value => type.toBuffer(value));
    const converted = buffers.map(buffer => type.fromBuffer(buffer));

    expect(buffers[0]).toEqual(buffers[1]);
    expect(converted).toEqual(expected);
    expect(buffers).toMatchSnapshot();
  });

  it('Test resolve from int', () => {
    const type = Type.forSchema(
      {
        type: 'record',
        fields: [{ name: 'startedAt', type: { type: 'int', logicalType: 'date' } }],
      } as Schema,
      { logicalTypes: { date: DateAsDateType } },
    );

    const longType = Type.forSchema({
      type: 'record',
      fields: [{ name: 'startedAt', type: 'int' }, { name: 'name', type: 'string' }],
    } as Schema);

    const expected = { startedAt: new Date('2018-06-05T00:00:00.000Z') };
    const longBuffer = longType.toBuffer({ startedAt: 17687, name: 'test' });
    const resolver = type.createResolver(longType);
    const value = type.fromBuffer(longBuffer, resolver);

    expect(value).toEqual(expected);
  });

  it('Test invalid resolver', () => {
    const type = Type.forSchema(
      {
        type: 'record',
        fields: [{ name: 'startedAt', type: { type: 'int', logicalType: 'date' } }],
      } as Schema,
      { logicalTypes: { date: DateAsDateType } },
    );

    const longType = Type.forSchema({
      type: 'record',
      fields: [{ name: 'startedAt', type: 'string' }],
    } as Schema);

    expect(() => type.createResolver(longType)).toThrowError();
  });
});
