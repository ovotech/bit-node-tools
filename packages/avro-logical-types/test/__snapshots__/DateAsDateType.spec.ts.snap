// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test DateAsDateType 1`] = `
Array [
  Object {
    "data": Array [
      174,
      148,
      2,
    ],
    "type": "Buffer",
  },
  Object {
    "data": Array [
      174,
      148,
      2,
    ],
    "type": "Buffer",
  },
]
`;
